src/printer.js

```diff
@@ -1203,7 +1203,13 @@ function genericPrintNoParens(path, options, print) {
           (n.value.type === "StringLiteral" || n.value.type === "Literal") &&
           typeof n.value.value === "string"
         ) {
-          res = '"' + util.htmlEscapeInsideDoubleQuote(n.value.value) + '"';
+          // Whenever the parser wouldn't convert it to an entity
+          // do not escape.
+          if (isParserKeepingEntities(n, options)) {
```

Comment:
As you told me, I use this helper function in order to check if the original value is different or not from what the parser returns.